src/Symfony/Bridge/Twig/Extension/CodeExtension.php

```diff
@@ -130,27 +125,36 @@ public function formatArgsAsText($args)
     /**
      * Returns an excerpt of a code file around the given line number.
      *
-     * @param string $file A file path
-     * @param int    $line The selected line number
+     * @param string $file       A file path
+     * @param int    $line       The selected line number
+     * @param int    $srcContext The number of displayed lines around or -1 for the whole file
      *
      * @return string An HTML string
      */
-    public function fileExcerpt($file, $line)
+    public function fileExcerpt($file, $line, $srcContext = 3)
     {
         if (is_readable($file)) {
             // highlight_file could throw warnings
             // see https://bugs.php.net/bug.php?id=25725
             $code = @highlight_file($file, true);
             // remove main code/span tags
             $code = preg_replace('#^<code.*?>\s*<span.*?>(.*)</span>\s*</code>#s', '\\1', $code);
-            $content = preg_split('#<br />#', $code);
+            // split multiline spans
+            $code = preg_replace_callback('#<span ([^>]++)>((?:[^<]*+<br \/>)++[^<]*+)</span>#', function ($m) {
+                return "<span $m[1]>".str_replace('<br />', "</span><br /><span $m[1]>", $m[2]).'</span>';
+            }, $code);
+            $code = explode('<br />', $code);
```

Comment:
please don't replace the ``$code`` variable here. Switching an existing variables from a string to an array makes the code harder to read. The array was named ``$content`` previously (``$lines`` might be a better name btw)